api/src/main/java/io/cloudevents/extensions/ExtensionsParser.java

```diff
@@ -0,0 +1,41 @@
+package io.cloudevents.extensions;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.Extension;
+
+import java.util.HashMap;
+import java.util.function.Supplier;
+
+public final class ExtensionsParser {
+
+    private static class SingletonContainer {
+        private final static ExtensionsParser INSTANCE = new ExtensionsParser();
```

Comment:
`ExtensionsParser` as `enum` with a single `INSTANCE` value would work as well

Reply:
TBH I prefer this way, the singletons with enums sounds too weird to me :smile:

Reply:
Actually... is it possible to avoid singletons for extensions by passing an extension parser to the event parser

It would be nice to be able to provide non-static extension parsers (e.g. by injecting with some DI framework)

Reply:
I think as a first pass is fine, we'll improve this part later. For now i just made the bare minimum to make it working, since it's my priority :smile:.